lib/utils.ts

```diff
@@ -0,0 +1,128 @@
+// Most of these util functions are copied as-is from https://github.com/openpgpjs/openpgpjs/blob/v5.0.0/src/util.js
+// @ts-ignore missing TS defs
+import { concatUint8Array, transform } from '@openpgp/web-stream-tools';
```

Comment:
It would be best not to import `@openpgp/web-stream-tools` for this, if possible, as it also has a dependency on the stream polyfills. I think you could copy / adapt `concatUint8Array` from there; and then for `transform`, is the stream support of `stringToUtf8Array` and `utf8ArrayToString` used anywhere? If not, you can remove the `transform`s just do it in one step.

If it *is* used, it won't work like this, you need to call `loadStreamsPonyfill()` in v0.0.10 of web-stream-tools before using a `TransformStream`, as otherwise it won't be available in all browsers.

Another option would be to manually create a `ReadableStream` that when read, reads the original stream and processes each chunk manually. Lmk if you want help with that.

Reply:
Even if it isn't used in the library, since pmcrypto supports streaming, I think we should have these utils support streams too... But as far as polyfilling goes, I'd let the client load them , it's what e.g. Drive already did.

Wouldn't using a ReadableStream still require an import?

Reply:
Let's go for the ReadableStreams then, to avoid making the function `async`